docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -43,13 +43,40 @@ OR
 }
 ```
 
-`"null"` is not considered for type casting logic, only for the nullability constraint. Type lookup will continue down the table.
+`"null"` is not considered for type casting logic, only for the nullability constraint. Type lookup will continue down
+the table. `"null"`'s position in a list does not matter. Nullability constraints only apply to the first major of the 
+schema. For example, `2-0-0` would use them, but `2-0-1` or `2-2-0` would not. And `3-0-0` would apply constraints again.
+It is still valid to have `"null"` for validation purposes in enrich.
 
 </td>
 <td>
 
 `NULLABLE`
 
+</td>
+</tr>
+<tr>
+<td >
+
+```json
+{
+    "properties": {
+      "f1": {"type": "T"}
+    },
+    "required": [ "f1" ]
+}
+```
+
+`"required"` is not considered for type casting logic, only for the nullability constraint. Type lookup will continue down the table.
+Fields that are not listed as `"required"` are nullable. Nullability constraints only apply to the first major of the
```

Comment:
Could you add an explanation of what happens when a field is marked as required but has `null` as one of the types or enum values?